src/testdir/test_ins_complete.vim

```diff
@@ -2289,6 +2344,16 @@ func Test_complete_info_index()
   call feedkeys("Go\<C-X>\<C-P>\<F5>\<Esc>_dd", 'tx')
   call assert_equal(-1, g:compl_info['selected'])
 
+  " Check if index out of range
+  " https://github.com/vim/vim/pull/12971
+  call feedkeys("Go\<C-X>\<C-N>\<C-P>\<F5>\<Esc>_dd", 'tx')
+  call assert_equal(0, g:compl_info['selected'])
```

Comment:
though test passed, but i donot understand why here the selected is 0?
looks this made 'noselect' much special.

Reply:
what do you mean? It's index `0` from the bottom. that should be correct, no?

Reply:
i think it is 5, which same like non-noselect, and same like if it's not around.
// but if it is the used way (as someone said `i donot know why was done that way`), then to be compatible, let it be.